infra/resources/modules/commons/appgateway.tf

```
@@ -14,4 +14,292 @@ resource "azurerm_public_ip" "appgateway_public_ip" {
   zones               = [1, 2, 3]
 
   tags = var.tags
+}
+
+# Subnet to host the application gateway
+module "appgateway_snet" {
+  source                                    = "git::https://github.com/pagopa/terraform-azurerm-v3.git//subnet?ref=v8.20.0"
+  name                                      = format("%s-appgateway-snet", local.project)
+  address_prefixes                          = var.cidr_subnet_appgateway
+  resource_group_name                       = azurerm_resource_group.rg_external.name
+  virtual_network_name                      = module.vnet.name
+  private_endpoint_network_policies_enabled = true
+
+  service_endpoints = [
+    "Microsoft.Web",
+  ]
+}
+
+## Application gateway ##
+module "app_gw" {
+  source = "github.com/pagopa/terraform-azurerm-v3.git//app_gateway?ref=v8.20.0"
```

```suggestion
  source = "github.com/pagopa/terraform-azurerm-v3//app_gateway?ref=v8.20.0"
```

If we don't use the module:
- do we need to write lot of code?
- do we still have the problem of random tf plans?